src/Caching/Hybrid/src/HybridCacheServiceExtensions.cs

```diff
@@ -0,0 +1,64 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Caching.Hybrid.Internal;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.Extensions.Internal;
+
+namespace Microsoft.Extensions.Caching.Hybrid;
+
+/// <summary>
+/// Configuration extension methods for <see cref="HybridCache"/>
+/// </summary>
+public static class HybridCacheServiceExtensions
+{
+    /// <summary>
+    /// Adds support for multi-tier caching services
+    /// </summary>
+    /// <returns>A builder instance that allows further configuration of the <see cref="HybridCache"/> system</returns>
+    public static IHybridCacheBuilder AddHybridCache(this IServiceCollection services, Action<HybridCacheOptions> setupAction)
+    {
+#if NET7_0_OR_GREATER
+        ArgumentNullException.ThrowIfNull(setupAction);
+#else
+        _ = setupAction ?? throw new ArgumentNullException(nameof(setupAction));
+#endif
+        AddHybridCache(services);
+        services.Configure(setupAction);
+        return new HybridCacheBuilder(services);
+    }
+
+    /// <summary>
+    /// Adds support for multi-tier caching services
+    /// </summary>
+    /// <returns>A builder instance that allows further configuration of the <see cref="HybridCache"/> system</returns>
+    public static IHybridCacheBuilder AddHybridCache(this IServiceCollection services)
+    {
+#if NET7_0_OR_GREATER
+        ArgumentNullException.ThrowIfNull(services);
+#else
+        _ = services ?? throw new ArgumentNullException(nameof(services));
+#endif
+
+#if NET8_0_OR_GREATER
+        services.TryAddSingleton(TimeProvider.System);
+#else
+        services.TryAddSingleton<ISystemClock, SystemClock>();
+#endif
+        services.AddOptions();
+        services.AddMemoryCache();
+        services.AddDistributedMemoryCache(); // we need a backend; use in-proc by default
+        services.AddSingleton<IHybridCacheSerializerFactory, DefaultJsonSerializerFactory>();
```

Comment:
Should these all be `TryAdd*`?

Reply:
they internally all use `TryAdd...`, but the public methods are `Add...`, so: this is "try"

Reply:
I believe `Add` will override any other versions of the interface that are registered. Whereas `TryAdd` will only add if no-one else has registered an implementation yet. Why we prefer `TryAdd` sometimes is when we would like users to be able to provide their own implementation and us not overriding that.

Reply:
@BrennanConroy understood, and again: all of these *are* `TryAdd`;

- https://github.com/dotnet/runtime/blob/324bd561b9e9d01ded8d2cb086cf55d23fc37eab/src/libraries/Microsoft.Extensions.Caching.Memory/src/MemoryCacheServiceCollectionExtensions.cs#L22-L30
- https://github.com/dotnet/runtime/blob/324bd561b9e9d01ded8d2cb086cf55d23fc37eab/src/libraries/Microsoft.Extensions.Caching.Memory/src/MemoryCacheServiceCollectionExtensions.cs#L22-L30
- https://github.com/dotnet/runtime/blob/324bd561b9e9d01ded8d2cb086cf55d23fc37eab/src/libraries/Microsoft.Extensions.Caching.Memory/src/MemoryCacheServiceCollectionExtensions.cs#L66-L74

Reply:
ah, I was focusing on the top 3 (`AddOptions` etc) which *are* `TryAdd`; the bottom few? yup, we can flip that